File: backend/test_auth.py
import requests
import json

# Configuration
BASE_URL = "http://localhost:5000"
TEST_USER = {
    "email_or_username": "test@example.com",  # Change to your test user
    "password": "password123"                 # Change to your test password
}

def test_auth_flow():
    print("Testing authentication flow...")
    
    # Step 1: Login to get token
    print("\n1. Logging in...")
    login_response = requests.post(
        f"{BASE_URL}/api/auth/login",
        headers={"Content-Type": "application/json"},
        data=json.dumps(TEST_USER)
    )
    
    if login_response.status_code != 200:
        print(f"❌ Login failed: {login_response.status_code}")
        print(login_response.text)
        return
    
    login_data = login_response.json()
    access_token = login_data.get("access_token")
    
    if not access_token:
        print("❌ No access token in response")
        print(login_data)
        return
    
    print(f"✅ Login successful, got token: {access_token[:10]}...")
    
    # Step 2: Test health endpoint (no auth required)
    print("\n2. Testing health endpoint...")
    health_response = requests.get(f"{BASE_URL}/api/health")
    
    if health_response.status_code != 200:
        print(f"❌ Health check failed: {health_response.status_code}")
    else:
        print("✅ Health check successful")
    
    # Step 3: Test profile endpoint (auth required)
    print("\n3. Testing profile endpoint (with auth)...")
    profile_response = requests.get(
        f"{BASE_URL}/api/auth/profile",
        headers={
            "Authorization": f"Bearer {access_token}",
            "Content-Type": "application/json"
        }
    )
    
    if profile_response.status_code != 200:
        print(f"❌ Profile check failed: {profile_response.status_code}")
        print(profile_response.text)
    else:
        print("✅ Profile check successful")
    
    # Step 4: Test dream analysis endpoint (auth required)
    print("\n4. Testing dream analysis endpoint...")
    dream_data = {
        "dreamText": "I was flying over mountains and oceans."
    }
    
    dream_response = requests.post(
        f"{BASE_URL}/api/dreams/analyze",
        headers={
            "Authorization": f"Bearer {access_token}",
            "Content-Type": "application/json"
        },
        data=json.dumps(dream_data)
    )
    
    if dream_response.status_code != 200:
        print(f"❌ Dream analysis failed: {dream_response.status_code}")
        print(dream_response.text)
    else:
        print("✅ Dream analysis successful")
        print("Response preview:")
        response_data = dream_response.json()
        print(f"- Dream ID: {response_data.get('dream_id')}")
        print(f"- Analysis starts with: {response_data.get('analysis', '')[:50]}...")

if __name__ == "__main__":
    test_auth_flow() 

File: lib/test_auth.dart
import 'dart:convert';
import 'package:flutter/material.dart';
import 'package:http/http.dart' as http;

void main() {
  runApp(const TestAuthApp());
}

class TestAuthApp extends StatelessWidget {
  const TestAuthApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Auth Test',
      theme: ThemeData(
        primarySwatch: Colors.purple,
        useMaterial3: true,
      ),
      home: const TestAuthScreen(),
    );
  }
}

class TestAuthScreen extends StatefulWidget {
  const TestAuthScreen({super.key});

  @override
  State<TestAuthScreen> createState() => _TestAuthScreenState();
}

class _TestAuthScreenState extends State<TestAuthScreen> {
  final TextEditingController _usernameController = TextEditingController();
  final TextEditingController _passwordController = TextEditingController();
  final TextEditingController _dreamTextController = TextEditingController();
  
  String _token = '';
  String _result = '';
  bool _isLoading = false;
  
  static const String baseUrl = 'http://localhost:5000';
  
  @override
  void dispose() {
    _usernameController.dispose();
    _passwordController.dispose();
    _dreamTextController.dispose();
    super.dispose();
  }
  
  Future<void> _login() async {
    setState(() {
      _isLoading = true;
      _result = 'Logging in...';
    });
    
    try {
      final response = await http.post(
        Uri.parse('$baseUrl/api/auth/login'),
        headers: {'Content-Type': 'application/json'},
        body: jsonEncode({
          'email_or_username': _usernameController.text,
          'password': _passwordController.text,
        }),
      );
      
      final data = jsonDecode(response.body);
      
      if (response.statusCode == 200) {
        setState(() {
          _token = data['access_token'];
          _result = 'Login successful!\nToken: ${_token.substring(0, 10)}...\n\n${jsonEncode(data)}';
        });
      } else {
        setState(() {
          _result = 'Login failed: ${data['message']}\n\n${jsonEncode(data)}';
        });
      }
    } catch (e) {
      setState(() {
        _result = 'Error: $e';
      });
    } finally {
      setState(() {
        _isLoading = false;
      });
    }
  }
  
  Future<void> _healthCheck() async {
    setState(() {
      _isLoading = true;
      _result = 'Checking health...';
    });
    
    try {
      final response = await http.get(Uri.parse('$baseUrl/api/health'));
      final data = jsonDecode(response.body);
      
      setState(() {
        _result = 'Health check: ${response.statusCode}\n\n${jsonEncode(data)}';
      });
    } catch (e) {
      setState(() {
        _result = 'Error: $e';
      });
    } finally {
      setState(() {
        _isLoading = false;
      });
    }
  }
  
  Future<void> _analyzeDream() async {
    if (_token.isEmpty) {
      setState(() {
        _result = 'Please login first!';
      });
      return;
    }
    
    setState(() {
      _isLoading = true;
      _result = 'Analyzing dream...';
    });
    
    try {
      final response = await http.post(
        Uri.parse('$baseUrl/api/dreams/analyze'),
        headers: {
          'Content-Type': 'application/json',
          'Authorization': 'Bearer $_token',
        },
        body: jsonEncode({
          'dreamText': _dreamTextController.text,
        }),
      );
      
      final data = jsonDecode(response.body);
      
      setState(() {
        _result = 'Analysis result (${response.statusCode}):\n\n${jsonEncode(data)}';
      });
    } catch (e) {
      setState(() {
        _result = 'Error: $e';
      });
    } finally {
      setState(() {
        _isLoading = false;
      });
    }
  }
  
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text('Auth Test'),
      ),
      body: SingleChildScrollView(
        padding: const EdgeInsets.all(16.0),
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.stretch,
          children: [
            const Text('Login', style: TextStyle(fontSize: 18, fontWeight: FontWeight.bold)),
            const SizedBox(height: 8),
            TextField(
              controller: _usernameController,
              decoration: const InputDecoration(
                labelText: 'Email or Username',
                border: OutlineInputBorder(),
              ),
            ),
            const SizedBox(height: 8),
            TextField(
              controller: _passwordController,
              decoration: const InputDecoration(
                labelText: 'Password',
                border: OutlineInputBorder(),
              ),
              obscureText: true,
            ),
            const SizedBox(height: 16),
            ElevatedButton(
              onPressed: _isLoading ? null : _login,
              child: const Text('Login'),
            ),
            const SizedBox(height: 16),
            ElevatedButton(
              onPressed: _isLoading ? null : _healthCheck,
              child: const Text('Health Check (No Auth)'),
            ),
            const SizedBox(height: 24),
            const Text('Dream Analysis', style: TextStyle(fontSize: 18, fontWeight: FontWeight.bold)),
            const SizedBox(height: 8),
            TextField(
              controller: _dreamTextController,
              decoration: const InputDecoration(
                labelText: 'Dream Text',
                border: OutlineInputBorder(),
              ),
              maxLines: 5,
            ),
            const SizedBox(height: 16),
            ElevatedButton(
              onPressed: _isLoading ? null : _analyzeDream,
              child: const Text('Analyze Dream'),
            ),
            const SizedBox(height: 24),
            const Text('Result:', style: TextStyle(fontSize: 18, fontWeight: FontWeight.bold)),
            const SizedBox(height: 8),
            Container(
              padding: const EdgeInsets.all(8),
              decoration: BoxDecoration(
                border: Border.all(color: Colors.grey),
                borderRadius: BorderRadius.circular(4),
              ),
              child: _isLoading
                ? const Center(child: CircularProgressIndicator())
                : SelectableText(_result, style: const TextStyle(fontFamily: 'monospace')),
            ),
          ],
        ),
      ),
    );
  }
} 